ingest-app/src/cmr/ingest/services/granule_bulk_update/s3/echo10.clj

```diff
@@ -110,16 +125,29 @@
 (defn- add-s3-url-to-metadata
   "Takes the granule ECHO10 xml and a list of S3 urls.
   Update the ECHO10 granule metadata with the S3 urls.
-  Returns the updated metadata."
-  [gran-xml urls]
+  Returns the updated metadata.
+
+  Valid operators are
+  :replace
+  :append"
+  [gran-xml urls op]
   (let [parsed (xml/parse-str gran-xml)
         online-accesses (xml-elem->online-access-urls parsed)]
-    (xml/indent-str (add-s3-url* parsed online-accesses urls))))
+    (xml/indent-str (add-s3-url* parsed online-accesses urls op))))
 
 (defn add-s3-url
   "Takes the ECHO10 granule concept and a list of S3 urls.
   Update the ECHO10 granule metadata with the S3 urls.
   Returns the granule concept with the updated metadata."
   [concept urls]
-  (let [updated-metadata (add-s3-url-to-metadata (:metadata concept) urls)]
+  (let [updated-metadata (add-s3-url-to-metadata (:metadata concept) urls :replace)]
+    (assoc concept :metadata updated-metadata)))
+
+(defn append-s3-url
+  "Append the ECHO10 granule metadata with the S3 urls.
+  Existing URLs will be preserved. If an existing URL is included in
```

Comment:
`Existing URLs will be preserved.` and `If an existing URL is included in the list of updates, it will be updated.` are contradicting statements.